src/Common.php

```diff
@@ -119,6 +120,10 @@ private static function createDefaultDecider(int $maxRetries): Closure
                 return true;
             } elseif ($error && $error->getCode() >= 500) {
                 return true;
+            } elseif ($error && (is_a($error, RequestException::class) || is_a($error, ConnectException::class))
+                && in_array($error->getHandlerContext()['errno'] ?? 0, [CURLE_RECV_ERROR, CURLE_SEND_ERROR])
+            ) {
```

Comment:
Tady bych to mozna jen trosku preformatoval pro citelnost, aby kazda root podminka byla na samostanym radku

```suggestion
            } elseif ($error &&
                 (is_a($error, RequestException::class) || is_a($error, ConnectException::class)) &&
                  in_array($error->getHandlerContext()['errno'] ?? 0, [CURLE_RECV_ERROR, CURLE_SEND_ERROR])
            ) {
```